src/Browser.php

```diff
@@ -725,6 +726,101 @@ public function withResponseBuffer($maximumSize)
         ));
     }
 
+    /**
+     * #### withHeader()
+     *
+     * The `withHeader(string $header, string $value): Browser` method can be used to
+     * add another header.
+     * For example:
+     *
+     * ```php
+     * $browser = $browser->withHeader('User-Agent', 'ACMC');
+     *
+     * $browser->get($url)->then(function (Psr\Http\Message\ResponseInterface $response) {
+     * // case-insensitive
+     * var_dump($response->getHeader('user-agent'));
+     * }, function (Exception $e) {
+     * echo 'Error: ' . $e->getMessage() . PHP_EOL;
+     * });
+     * ```
+     *
+     * Note that the new header will (case insensitiv) overwrite an existing header with the same name.
+     * Headers passed to the HTTP methods will be preferred.
+     *
+     * ```php
+     * $browser = $browser->withHeader('User-Agent', 'ACMC');
+     *
+     * $browser->get($url, array('user-agent' => 'value'))->then(function (Psr\Http\Message\ResponseInterface $response) {
+     * // case-insensitive
+     * var_dump($response->getHeaders());
+     * //will return array(
+     * //                    ...
+     * //                    'user-agent' => 'value'
+     * //                    ...
+     * //                 )
+     * }, function (Exception $e) {
+     * echo 'Error: ' . $e->getMessage() . PHP_EOL;
+     * });
+     * ```
+     *
+     * @param string $header
+     * @param string $value
+     * @return Browser
+     */
+    public function withHeader($header, $value)
+    {
+        $browser = clone $this;
+        $this->removeDuplicatesFromDefaultHeaders($browser, $header);
```

Comment:
It's interesting this is in fact `withoutHeader()`, perhaps we should move the entire logic there and reuse the method here?

Reply:
did it